wwexport/ww_html.py

```diff
@@ -83,17 +83,27 @@ def _jinja_filter_name_case(val: str):
 SPACE_MENTION_RE = r"(<)(@space)>"
 STRONG_RE = r"(\*)(.+?)\*"
 
+def file_path_for_id(id):
+    if id in paths:
+        return "file://" + paths[id]
+    else:
+        try:
+            short_id = id.split("@")[-1]
+            if short_id in paths:
+                return "file://" + paths[short_id]
+        except IndexError:
+            logger.error("Could not resolve link in message to file %s, ID is an unexpected format", id)
+
+        logger.error("Could not resolve link in message to file %s", id)
+        return ""
+
 class FilePattern(InlineProcessor):
     def handleMatch(self, m, data):
         global paths
```

Comment:
since `paths` isn't referenced directly in either FilePattern or ImagePattern it probably makes sense to move this to `file_path_for_id` for clarity

Reply:
Thanks - good catch! I removed the two global declarations. It's actually not needed in the new method (and wasn't really needed in the old method). It's only needed for assignments. I may try, in a later PR, to refactor to get rid of the global anyway - by having the functions declared in a nested context so they can reference a non-local but not necessarily global paths variable, but that will come later if I get to it.